src/rust/ide/src/model/suggestion_database.rs

```diff
@@ -46,36 +47,46 @@ pub struct Entry {
 
 impl Entry {
     /// Create entry from the structure deserialized from the Language Server responses.
-    pub fn from_ls_entry(entry:language_server::types::SuggestionEntry) -> FallibleResult<Self> {
+    pub fn from_ls_entry(entry:language_server::types::SuggestionEntry, logger:impl AnyLogger)
+        -> FallibleResult<Self> {
         use language_server::types::SuggestionEntry::*;
+        let convert_doc = |doc: Option<String>| doc.and_then(|doc| match Entry::gen_doc(doc) {
+            Ok(d)  => Some(d),
+            Err(err) => {
```

Comment:
there are functions to convert Result to Option, please use them instead, it will be shorter

Reply:
Yes there are - like `.ok()` or `.err()`, but they're unusable in our case, as we want to log the error, which would be impossible with `.ok` and `.err` would lose the value.
Here - all of available transformations : https://docs.google.com/drawings/d/1EOPs0YTONo_FygWbuJGPfikO9Myt5HwtiFUHRuE1JVM/preview